core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -59,13 +60,14 @@ private BootstrapCheck() {
      * checks
      *
      * @param settings              the current node settings
+     * @param dataDirectoryNumber   the integer number of the directories who's locks under the data directory we were able to take
      * @param boundTransportAddress the node network bindings
      */
-    static void check(final Settings settings, final BoundTransportAddress boundTransportAddress) {
+    static void check(final Settings settings, final int dataDirectoryNumber, final BoundTransportAddress boundTransportAddress) {
```

Comment:
This doesn't feel right to me, adding a parameter to the bootstrap check infrastructure that is specific to a check. I think that we should try to find a different approach (I'm happy to help brainstorm about this, but I have not yet done so).